### PR TITLE
[SPARK-39521][INFRA] Separate scheduled jobs to each workflow

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to separate workflows for each scheduled jobs. After this PR, each scheduled build can be easily searched at https://github.com/apache/spark/actions. For example, as below:

Screen Shot 2022-06-20 at 6 01 04 PM

In addition, this PR switches ANSI build to a scheduled build too because it's too expensive to run it for each commit.

### Why are the changes needed?

Currently it is very inconvenient to navigate scheduled jobs now. We should use https://github.com/apache/spark/actions/workflows/build_and_test.yml?query=event%3Aschedule link and manually search one by one.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Tested in my fork (https://github.com/HyukjinKwon/spark/actions).

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -23,102 +23,32 @@ on:
...
+        default: 8
+      branch:
+        required: false
+        type: string
+        default: master
+      hadoop:
+        required: false
+        type: string
+        default: hadoop3
+      type:
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:40]: I'll try to get rid of this argument in SPARK-39529

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun @Yikun @gengliangwang @MaxGekk FYI

**Comment by HyukjinKwon** [comment]
Seems working fine as expected.

We can merge this once https://github.com/HyukjinKwon/spark/actions/runs/2527942990 gets passed.

After this PR, developers should fetch and rebase their mater branch. I will send an PSA email to the dev mailing lists.

Other old branches won't be affected if I am not mistaken.

**Comment by HyukjinKwon** [comment]
Seems working fine as expected.

We can merge this once https://github.com/HyukjinKwon/spark/actions/runs/2527981545 gets passed.

After this PR, developers should fetch and rebase their mater branch. I will send an PSA email to the dev mailing lists.

Other old branches won't be affected if I am not mistaken.

**Comment by HyukjinKwon** [comment]
Guys I'll merge this soon and monitor today (it's morning my time) so I can follow up and fix some more if there are something going wrong

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
+1, LGTM!

**Comment by cloud-fan** [comment]
great job!

File: .github/workflows/build_ansi.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "Build / ANSI (master, Hadoop 3, JDK 8, Scala 2.12)"
```
Comment (cloud-fan) [inline .github/workflows/build_ansi.yml:20]: to clarify, we only test ansi mode in the master branch, not 3.3?

Comment (HyukjinKwon) [reply .github/workflows/build_ansi.yml:20]: BTW, I think we can set more scheduled jobs. At least roughly 2x schduled jobs can be bearable given my observations

**Comment by gengliangwang** [comment]
Late LGTM, thanks for the work!

**Pull request closed without merging**